### ISSUE TITLE
State migration smoke test

### ISSUE THREAD

**Issue by vancluever** [vancluever/terraform-provider-acme#49]
Just want to make sure that a state migration works:

- [x] Create a reg/cert using v0.6.0
- [x] Swap in the 1.0.0 provider
- [x] Validate that IDs have not changed on refresh
- [x] Crank `min_days_remaining` and renew cert

**Comment by vancluever** [comment]
`terraform refresh` after swapping in the new binary and updating the config gave this:

```
* acme_registration.reg: acme_registration.reg: directory missing new registration URL
```

This is, of course, because of the fact that a `terraform refresh` does not grab from new config.

This is actually making a case for something that I had been thinking about for while, which was the fact that the `server_url` should probably be a provider-level configuration variable. This was not considered as I wanted to make sure that the upgrade was as clean as possible, to save time on everyone's part.

In order to facilitate upgrades though, we might need to actually take this on.

**Comment by vancluever** [comment]
This works as expected after the fixes in #50.
